widgets/text/text.go

```diff
@@ -109,6 +109,17 @@ func (t *Text) Write(text string, wOpts ...WriteOption) error {
 		t.reset()
 	}
 	for _, r := range text {
+		// Implement basic buffer limit to avoid excess memory issues
+		if len(t.content) > 1 && len(t.content) >= t.opts.maxContent && t.opts.maxContent > 0 {
+			// If the new content is longer than all existing, simply reset
+			if len(string(r)) >= len(t.content) {
```

Comment:
I may be missing something, but since `r` is a single rune in this for loop, doesn't the expression `len(string(r))` always return 1?

Reply:
I've moved the logic to be outside of the stepping through the runes within the string so it should make more sense again.